eventnotifier/http/http.go

```diff
@@ -126,27 +129,34 @@ func (n *Notifier) Notify(dis v1beta1.Disruption, event corev1.Event, notifType
 		}
 	}
 
+	disruptionJSON, err := json.Marshal(dis)
+	if err != nil {
+		return fmt.Errorf("http notifier: couldn't marshal disruption: %s", err)
+	}
+
 	notif := HTTPNotifierEvent{
+		ID:                 string(dis.UID),
 		NotificationTitle:  utils.BuildHeaderMessageFromDisruptionEvent(dis, notifType),
 		NotificationType:   notifType,
 		EventMessage:       utils.BuildBodyMessageFromDisruptionEvent(dis, event, false),
 		InvolvedObjectKind: dis.Kind,
 		DisruptionName:     dis.Name,
+		Disruption:         string(disruptionJSON),
 		Cluster:            n.common.ClusterName,
 		Namespace:          dis.Namespace,
 		TargetsCount:       len(dis.Status.TargetInjections),
 		Username:           emailAddr.Name,
 		UserEmail:          emailAddr.Address,
 	}
 
-	body, err := json.Marshal(notif)
-	if err != nil {
-		return fmt.Errorf("http notifier: couldn't send notification: %s", err.Error())
+	body := bytes.NewBuffer(nil)
+	if err := jsonapi.MarshalOnePayloadEmbedded(body, &notif); err != nil {
```

Comment:
Curious to know; what's the benefit of the `jsonapi` library in place of `json` ? Does it format things better for API calls ?

Reply:
From what I understand, `jsonapi` structures responses to efficiently manage data relationships and minimize bandwidth, while plain `json` is simpler but lacks these optimizations. I think this article: [The Benefits of Using JSON API](https://nordicapis.com/the-benefits-of-using-json-api/) is a good read if you want a deeper explanation.